### ISSUE TITLE
Improve documentation

### ISSUE THREAD

**Issue by ffried** [CocoaLumberjack/CocoaLumberjack#960]
There are a few things that should be improved in the documentation:

[ ] Upgrade to new standards (iOS 12)
[ ] Add documentation for changes #952 (this one is on me)
[ ] Improve documentation for Swift
[ ] Maybe adjust documentation for #959 (/cc @bpoplauschi)

**Comment by bpoplauschi** [comment]
@ffried good list.
I would add
- [ ] look into deprecating `DDASLLogCapture` along with `DDTTYLogger` and `DDASLLogger` for the newer OS versions.

**Comment by ffried** [comment]
Good point. Although that requires some code changes first. But yea, we should also mention that in the docs then.

**Comment by bpoplauschi** [comment]
Linking #759 and #757 to the documentation updates needed

**Comment by lolgear** [comment]
[ ] use `__auto_type` inference in documentation. (?)

Yes, auto type inference has its own downsides ( time spent by type deducing ).
But wait, it is what whole Swift does every second. 
And yes, for examples it could be acceptable.
